### DISCUSSION TITLE
Send notification after validation

### DISCUSSION THREAD

**Discussion by mpestanasoares** [filamentphp/filament#4081] [Help] [answered]
I have a large form with validation rules on each field and I'm using Wizard Layout. In addition to highlighting the fields where validation fails, I would like to send a notification via Filament Notifications after click Next button. How do I do this?

**Comment by Z3d0X** [comment]
```php
Wizard::make([
    Wizard\Step::make('Order')
        ->schema([
            // ...
        ])
        ->afterValidated(fn () => Filament::notify('success', 'Step Complete')),
])

**Reply by danharrin** [reply]
I think he wants to display the validation error in the notification

**Comment by danharrin** [comment, marked as answer]
On the Create / Edit page class:

```php
protected function onValidationError(ValidationException $exception): void
{
    Notification::make()
        ->title($exception->getMessage())
        ->danger()
        ->send();
}
```

**Reply by mpestanasoares** [reply]
That's exactly what i want. Thanks!

**Reply by Z3d0X** [reply]
On the subject of this, is there a way to globally set this for all pages?

**Reply by danharrin** [reply]
```php
use Filament\Notifications\Notification;
use Filament\Pages\Page;

Page::$reportValidationErrorUsing = function (ValidationException $exception) {
    Notification::make()
        ->title($exception->getMessage())
        ->danger()
        ->send();
};
```

you can add it somewhere in the docs if you can work out where it should be

**Reply by intrepidws** [reply]
Does this method not work with custom pages?

**Reply by aymanalhattami** [reply]
How to handle exceptions using filament notifications i mean show notification instead of 500 error modal?

**Reply by danharrin** [reply]
In v4 https://filamentphp.com/docs/4.x/panel-configuration#configuring-error-notifications

**Reply by aymanalhattami** [reply]
Is there a way to handle it in v3?

**Reply by danharrin** [reply]
You could use a Livewire JS error hook, yes
